stdlib/public/core/CocoaArray.swift

```diff
@@ -51,12 +51,16 @@ internal struct _CocoaArrayWrapper: RandomAccessCollection {
 
   @usableFromInline
   internal var endIndex: Int {
-    return core.count
+    @_effects(releasenone) get {
```

Comment:
My claim is that any NSArray subclass that responds to -count or -objectAtIndex: by releasing something relevant is broken and will crash when used in ObjC code anyway